### PR TITLE
SpacemanDMM parser + other fixes

### PR DESCRIPTION
**Pull request by PJB3005**
See commits.

The spacemanDMM parser does not currently output working games yet. 

The existing parser is still kept in and is behind a compiler switch (`--old-parser`)

Changes to SpacemanDMM are at https://github.com/space-wizards/SpacemanDMM/tree/opendream

### CODE REVIEW

File: DMCompiler/SpacemanDmm/AstConverter.cs [hunk 1]
```diff
@@ -0,0 +1,853 @@
...
+                    {
+                        case NewType.Implicit:
+                            return new DMASTNewInferred(argsConverted);
+                        case NewType.MiniExpr miniExpr:
+                            var dereferences = miniExpr.Fields.Select(f =>
+                            {
+                                var (conditional, type) = ConvertPropertyAccessKind(f.Kind);
+                                return new DMASTDereference.Dereference(type, conditional, f.Ident);
+                            }).ToArray();
+                            return new DMASTDereference(new DMASTIdentifier(miniExpr.Ident), dereferences);
```
Comment (wixoaGit) [inline DMCompiler/SpacemanDmm/AstConverter.cs:663]:
```suggestion
                            var dereference = new DMASTDereference(new DMASTIdentifier(miniExpr.Ident), dereferences);

                            return new DMASTNewDereference(dereference, argsConverted);
```

This fixes an issue with TestGame that was preventing the player mob from being created.

File: DMCompiler/Program.cs [hunk 2]
```diff
@@ -16,100 +17,171 @@
...
+
+        private static void PrintHelp()
+        {
+            Console.WriteLine(@"usage: DMCompiler [options] <source to compile>
+Arguments:
+    source to compile       The .dme environment or .dm source file to compile.
 
-            return true;
+Options:
+    --old-parser            Use old parser instead of SpacemanDMM.
```
Comment (ike709) [inline DMCompiler/Program.cs:158]: Not mentioning `--dump-preprocessor`?

**Comment by ike709** [comment]
`initial()` doesn't work

**Comment by ike709** [comment]
This seems like it should just be closed at this point tbh.

**Pull request closed without merging**